### PR TITLE
Add get tx execution output rpc

### PR DESCRIPTION
**Pull request by musitdev**
Update get_transaction to return a json adapted redeable RPC type.
The specs:
```
Call:
get_transaction {
        //tx Hash to find 
        hash: String,
},

=> return the Json formated Tx as follow
[
    {   
        //hex encoded pubkey
        author: String,
        //hex encoded tx hash
        hash: String,
        //Payload of the Tx see RpcPayload
        payload: RpcPayload
        nonce: u64,
        //hex encoded signature
        signature: String,
    }

    "RpcPayload" : 
    {
        "Empty" : "",
        "Transfer" : {
            //hex encoded pk
            to: String,
            //String representation of the Bigint
            value: String,
        },
        "Stake" : {
            //String representation of the Bigint
            value: String,
        },
        "Unstake" : {
            //String representation of the Bigint
            value: String,
        },
        "Deploy" : {
            name: String,
            //Json serialized ProgramMetadata
            prover: ProgramMetadata,
            //Json serialized ProgramMetadata
            verifier: ProgramMetadata,
        },
        "Run" : {
            //Json serialized ProgramMetadata
            workflow: Workflow,
        },
        "Proof" : {
            //hex encoded hash
            parent: String,
            //hex encoded hash
            prover: String,
            //Base64 serialization of the proof
            proof: String,
            files: [
                //url of the file. File can be retrieve using the node HTTP server with the url.
                url: String,
                //checksum of the file for verification
                checksum: String,
                //Path of the file inside the VM. Use to help to recognize it.
                vm_path: String
            ],
        },
        "ProofKey" : {
            //hex encoded hash
            parent: String,
            key: Vec<u8>,
        },
        "Verification" : {
            //hex encoded hash
            parent: String,
            //hex encoded hash
            verifier: String,
            //Base64 serialization of the verification
            verification: Sting,
            files: [
                //url of the file. File can be retrieve using the node HTTP server with the url.
                url: String,
                //checksum of the file for verification
                checksum: String,
                //Path of the file inside the VM. Use to help to recognize it.
                vm_path: String
            ],
        },
        "Cancel" : {
            //hex encoded hash
            parent: String,
        },

    }
```

### CODE REVIEW

File: crates/node/src/types/rpc_types.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        host: SocketAddr,
+    ) -> Self {
+        //use parse to create an URL, no new method.
+        let mut url = reqwest::Url::parse(&format!("{}localhost", scheme))
+            .unwrap_or(reqwest::Url::parse("http://localhost").unwrap()); //unwrap always succeed
+        url.set_ip_host(host.ip()).unwrap(); //unwrap always succeed
+        url.set_port(Some(host.port())).unwrap(); //unwrap always succeed
+        url.set_path(&file.clone().into_download_file(tx_hash).get_uri());
+
+        println!("{url:?}",);
```
Comment (tuommaki) [inline crates/node/src/types/rpc_types.rs:33]:
```suggestion
        tracing::debug!("{url:?}",);
```
:question:

File: crates/node/src/types/transaction.rs [hunk 2]
```diff
@@ -250,6 +250,20 @@ impl Payload {
...
+    pub fn get_execution_output(self) -> Option<(Vec<TxFile<Output>>, Vec<u8>)> {
+        match self {
+            Payload::Proof {
+                proof: data, files, ..
+            }
+            | Payload::Verification {
+                verification: data,
+                files,
+                ..
+            } => Some((files, data)),
```
Comment (tuommaki) [inline crates/node/src/types/transaction.rs:263]: Should the return fields be maybe `(data, files)` when that's the order throughout the code base (including gRPC's `TaskResult`)?

File: crates/node/src/scheduler/mod.rs [hunk 3]
```diff
@@ -482,6 +482,13 @@ impl TaskManager for Scheduler {
                 running_task.task_started.elapsed().as_secs()
             );
 
+            if let Err(err) = self.database.mark_tx_executed(&running_task.task.tx).await {
+                tracing::error!(
+                    "failed to update transaction.executed => true - tx.hash: {}",
+                    &running_task.task.tx
+                );
+            }
+
```
Comment (tuommaki) [inline crates/node/src/scheduler/mod.rs:491]: I created a separate PR for this to get it incorporated ASAP: https://github.com/gevulotnetwork/gevulot/pull/114

File: crates/cli/Cargo.toml [hunk 4]
```diff
@@ -9,7 +9,7 @@ anyhow = "1.0.71"
 blake3 = { version = "1.5", features = [ "mmap" ] }
 clap = { version = "4.0.29", features = ["derive"] }
 futures-util = { version = "0.3", default-features = false }
-gevulot-node = { path = "../node" }
+gevulot-node = { path = "../node", default-features = false }
```
Comment (tuommaki) [inline crates/cli/Cargo.toml:12]: I extracted this to separate PR as it's needed ASAP for the blog post: https://github.com/gevulotnetwork/gevulot/pull/117

File: crates/cli/src/main.rs [hunk 5]
```diff
@@ -96,9 +96,22 @@ enum ConfCommands {
         #[clap(short, long, value_name = "TASK ARRAY")]
         tasks: String,
     },
+    /// Return the tree of executed tx associated to the specified Run Tx (Exec cmd).
+    #[command(arg_required_else_help = true)]
     PrintTxTree {
+        /// Hash of the Run Tx to look for.
+        #[clap(long, value_name = "TX HASH")]
```
[the remark this answers is not in the document]
Comment (musitdev) [reply crates/cli/src/main.rs:103]: Yes I agree. I copy / paste from another command, where I get some issue with -h for hash and help. I thought it was the same. I revert.

File: crates/node/src/rpc_server/mod.rs [hunk 6]
```diff
@@ -43,15 +46,20 @@ impl RpcServer {
         database: Arc<Database>,
         tx_sender: TxEventSender<RpcSender>,
     ) -> Result<Self> {
+        let mut local_http_host = cfg.p2p_listen_addr;
+        local_http_host.set_port(cfg.http_download_port);
```
[the remark this answers is not in the document]
Comment (musitdev) [reply crates/node/src/rpc_server/mod.rs:50]: Yes I change it.

File: crates/node/src/scheduler/mod.rs (hunk 3 above)
Comment (musitdev) [reply crates/node/src/scheduler/mod.rs:491]: Ok I see I'll rebase when merged

File: crates/cli/Cargo.toml (hunk 4 above)
Comment (musitdev) [reply crates/cli/Cargo.toml:12]: OK I see I'll rebase.

File: crates/node/src/main.rs [hunk 7]
```diff
@@ -253,7 +256,10 @@ async fn run(config: Arc<Config>) -> Result<()> {
     let workflow_engine = Arc::new(WorkflowEngine::new(database.clone()));
     let download_url_prefix = format!(
         "http://{}:{}",
-        config.p2p_listen_addr.ip(),
+        config
+            .p2p_advertised_listen_addr
+            .unwrap_or(p2p_listen_addr)
+            .ip(),
```
Comment (tuommaki) [inline crates/node/src/main.rs:262]: This looks weird. How come it doesn't conflict with `main`: https://github.com/gevulotnetwork/gevulot/blob/main/crates/node/src/main.rs#L257-L264 :thinking:

File: crates/node/src/networking/p2p/noise.rs [hunk 8]
```diff
@@ -173,7 +173,10 @@ impl Encoder<Bytes> for Codec {
     fn encode(&mut self, msg: Bytes, dst: &mut BytesMut) -> Result<(), Self::Error> {
         match self.noise {
             State::Handshake(ref mut noise) => {
-                let msg_len = noise.0.write_message(&msg, &mut self.buffer).unwrap();
+                let msg_len = noise
+                    .0
+                    .write_message(&msg, &mut self.buffer)
+                    .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
```
Comment (tuommaki) [inline crates/node/src/networking/p2p/noise.rs:179]: Same here. This "diff" is already in `main`: https://github.com/gevulotnetwork/gevulot/blob/main/crates/node/src/networking/p2p/noise.rs#L176-L179 :thinking:

**Review by tuommaki — APPROVED** [review]
LGTM :+1: 

Maybe merge latest `main` in first as some diffs looked a bit weird?

File: crates/node/src/main.rs (hunk 7 above)
Comment (musitdev) [reply crates/node/src/main.rs:262]: I've already rebase this commit. This modification came from it. Strange, it's still a diff.

File: crates/node/src/networking/p2p/noise.rs (hunk 8 above)
Comment (musitdev) [reply crates/node/src/networking/p2p/noise.rs:179]: The same.

Comment (musitdev) [reply crates/node/src/networking/p2p/noise.rs:179]: I've rebased from master. Some commit  came back. I think it's because individual commit are squash so rebase can't detect them.
I think it's ok, I've done some local test.
